### PR TITLE
Added crop image dialog to crop avatar

### LINKED ISSUE #3269: Feature to crop an avatar uploaded by the user
**Expected Behaviour**

Add a feature to crop an avatar uploaded by the user before saving changes.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Dilshaad21**
Fixes #3269 

Changes: 
- Added crop image dialog to crop an avatar uploaded by the user.

Demo Link: https://pr-3274-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 
final_5e1dd7bad16dff00167806d3_85795

**new changes**
Screenshot from 2020-01-15 18-22-00
Screenshot from 2020-01-15 18-24-56

### CODE REVIEW

**Comment by anishagg17** [comment]
Dialog is missing a cancel option/button

**Comment by Dilshaad21** [comment]
@akshatnitd PTAL.

File: src/components/Settings/AccountTab.react.js [hunk 1]
```diff
@@ -216,16 +223,45 @@ class AccountTab extends React.Component {
...
+        () => {
+          this.onImageCrop();
+        },
+      );
     };
     reader.readAsDataURL(file);
     this.handleMenuClose();
   };
 
+  onImageCrop = () => {
```
Comment (akshatnitd) [inline src/components/Settings/AccountTab.react.js]: Rename it something related to opening crop modal

File: src/components/Settings/AccountTab.react.js [hunk 2]
```diff
@@ -191,8 +195,11 @@ class AccountTab extends React.Component {
   };
 
   handleAvatarSubmit = async () => {
-    const { file } = this.state;
+    const { imageFile } = this.state;
     const { accessToken, actions, userEmailId } = this.props;
+    const fileExt = extractImageFileExtensionFromBase64(imageFile);
+    const fileName = 'image.' + fileExt;
+    const file = base64StringtoFile(imageFile, fileName);
```
Comment (Dilshaad21) [inline src/components/Settings/AccountTab.react.js]: @akshatnitd I have converted the image data(croppedImageData) which I get in `imageFile` in `base64` format to file format.
Then appended it using `form.append(''image",file)`. And the croppedImage is getting saved successfully.

File: src/components/shared/Dialog/dialogTypes/CropDialog.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      crop.y * scaleY,
+      crop.width * scaleX,
+      crop.height * scaleY,
+      0,
+      0,
+      crop.width,
+      crop.height,
+    );
+
+    let imageBase64 = canvas.toDataURL('image');
```
Comment (Dilshaad21) [inline src/components/shared/Dialog/dialogTypes/CropDialog.js]: `imageBase64`  contains the image data in `base64` format which is passed to `AccountTab.react.js` component using function `handleConfirm`.

**Comment by Dilshaad21** [comment]
@akshatnitd the cropped image is getting saved perfectly.

**Comment by Dilshaad21** [comment]
Reduced dialog size to `xs`, it will be fine now for all screensizes.

**Comment by Dilshaad21** [comment]
@akshatnitd Set UX to upload the avatar in the Modal itself and then save changes.

File: src/components/Settings/AccountTab.react.js [hunk 4]
```diff
@@ -257,7 +294,7 @@ class AccountTab extends React.Component {
       });
       return;
     }
-
+    // this.handleAvatarSubmit();
```
Comment (akshatnitd) [inline src/components/Settings/AccountTab.react.js]: Remove comment

File: src/components/shared/Dialog/index.js [hunk 5]
```diff
@@ -141,6 +143,7 @@ const DialogSection = props => {
     style = {
       padding: isMobileView() ? '0.3rem' : '2.4rem 1.5rem',
       textAlign: 'center',
+      // maxHeight: '100%',
```
Comment (akshatnitd) [inline src/components/shared/Dialog/index.js]: Remove comment

**Comment by akshatnitd** [comment]
Also update the branch

**Pull request merged**